alchemist-loading/src/main/kotlin/it/unibo/alchemist/boundary/LoadAlchemist.kt

```diff
@@ -28,47 +29,106 @@ object LoadAlchemist {
         ?.let { it.groupValues[1] }
         ?: error("Cannot extract package from ${T::class.jvmName}")
 
+    /**
+     * Load from an [input] [String] with overrides.
+     */
+    @JvmStatic
+    fun from(input: String, model: AlchemistModelProvider, overrides: List<String>) =
+        SimulationModel.fromMap(
+            applyOverrides(model.from(input), overrides),
+        )
+
     /**
      * Load from an [input] [String].
      */
     @JvmStatic
-    fun from(input: String, model: AlchemistModelProvider) = SimulationModel.fromMap(model.from(input))
+    fun from(input: String, model: AlchemistModelProvider) =
+        from(input, model, emptyList())
+
+    /**
+     * Load from a [reader] with overrides.
+     */
+    @JvmStatic
+    fun from(reader: Reader, model: AlchemistModelProvider, overrides: List<String>) =
+        SimulationModel.fromMap(
+            applyOverrides(model.from(reader), overrides),
+        )
 
     /**
      * Load from a [reader].
      */
     @JvmStatic
-    fun from(reader: Reader, model: AlchemistModelProvider) = SimulationModel.fromMap(model.from(reader))
+    fun from(reader: Reader, model: AlchemistModelProvider) =
```

Comment:
An alternative and more idiomatic kotlin approach would be to have a default value for overrides in the signature:

,overrides: List<String> = emptyList()

This however causes some issues when inter-oping with Java (when calling kotlin from java) because java doesn't automatically infer the default value and requires it to be passed.

Since we have some java tests calling into this class, I preferred this more verbose solution, but safer.

Reply:
This is a much better solution, I've updated the signatures